devito/equation.py

```diff
@@ -104,8 +113,13 @@ def _symbolic_functions(self):
 
     def xreplace(self, rules):
         """"""
-        return self.func(self.lhs.xreplace(rules), self.rhs.xreplace(rules),
-                         subdomain=self._subdomain)
+        return self.func(self.lhs.xreplace(rules), rhs=self.rhs.xreplace(rules),
+                         implicit_dims=self._implicit_dims, subdomain=self._subdomain)
+
+    def __str__(self):
+        return "%s(%s, %s)" % (self.__class__.__name__, self.lhs, self.rhs)
```

Comment:
Why is this necessary?

Reply:
Oh...not strictly. But I dropped the equivalent in Inc (see below), so that any future subclasses are stringified as 
```
Eq(...)
Inc(...)
NewSubClass(...)
```
if you look at the commit history of this PR, you see that at some point I had a new subclass of Eq called Step, then I removed it. But I might reinstate it in the future